electrum/lnchannel.py

```diff
@@ -851,15 +868,20 @@ def sweep_address(self) -> str:
         addr = None
         assert self.is_static_remotekey_enabled()
         our_payment_pubkey = self.config[LOCAL].payment_basepoint.pubkey
-        addr = make_commitment_output_to_remote_address(our_payment_pubkey)
-        if self.lnworker:
-            assert self.lnworker.wallet.is_mine(addr)
+        addr = make_commitment_output_to_remote_address(our_payment_pubkey, has_anchors=self.has_anchors())
+        # this assert fails with anchor output channels
+        #if self.lnworker:
+        #    assert self.lnworker.wallet.is_mine(addr)
         return addr
```

Comment:
Something weird happen in a rebase here.
The original branch looks like this: https://github.com/bitromortac/electrum/blob/5cb97c077c799aa6ebd06492645b9c4f738f85f8/electrum/lnchannel.py#L325
which looks okay.

However this new implementation is not good. There is a reason that assert is there, and it failing signals a problem.
The `sweep_address()` method is supposed to return a wallet address that we can send coins to as part of closing this channel. It is called from a lot of places, e.g. code that is trying to sweep HTLCs will send coins here, or even the mutual close will send our coins here. Clearly we don't want the mutual close to send coins to a CSV-locked non-wallet address :)

I've tried to clarify this method in https://github.com/spesmilo/electrum/commit/dd140df17dbba95108f1383c2e5d4c7c8d3e3b9e

Reply:
fixed in ccd4e2bcf08901a95a0a46d4367b8a10e8976d56